microceph/database/diskpath.go

```diff
@@ -1,37 +1,35 @@
-//go:generate mapper stmt -d github.com/canonical/microcluster/cluster -e disk delete-by-Member table=disks
-//go:generate mapper stmt -d github.com/canonical/microcluster/cluster -e disk delete-by-Member-and-Path table=disks
-//go:generate mapper stmt -d github.com/canonical/microcluster/cluster -e disk update table=disks
+//go:generate mapper stmt -d github.com/canonical/microcluster/cluster -e DiskPath objects table=diskpaths
+//go:generate mapper stmt -d github.com/canonical/microcluster/cluster -e DiskPath objects-by-Member table=diskpaths
+//go:generate mapper stmt -d github.com/canonical/microcluster/cluster -e DiskPath objects-by-Member-and-Path table=diskpaths
+//go:generate mapper stmt -d github.com/canonical/microcluster/cluster -e DiskPath id table=diskpaths
+//go:generate mapper stmt -d github.com/canonical/microcluster/cluster -e DiskPath create table=diskpaths
+//go:generate mapper stmt -d github.com/canonical/microcluster/cluster -e DiskPath delete-by-Member table=diskpaths
+//go:generate mapper stmt -d github.com/canonical/microcluster/cluster -e DiskPath delete-by-Member-and-Path table=diskpaths
+//go:generate mapper stmt -d github.com/canonical/microcluster/cluster -e DiskPath update table=diskpaths
 //
-//go:generate mapper method -i -d github.com/canonical/microcluster/cluster -e disk GetMany
-//go:generate mapper method -i -d github.com/canonical/microcluster/cluster -e disk GetOne
-//go:generate mapper method -i -d github.com/canonical/microcluster/cluster -e disk ID
-//go:generate mapper method -i -d github.com/canonical/microcluster/cluster -e disk Exists
-//go:generate mapper method -i -d github.com/canonical/microcluster/cluster -e disk Create
-//go:generate mapper method -i -d github.com/canonical/microcluster/cluster -e disk DeleteOne-by-Member-and-Path
-//go:generate mapper method -i -d github.com/canonical/microcluster/cluster -e disk DeleteMany-by-Member
-//go:generate mapper method -i -d github.com/canonical/microcluster/cluster -e disk Update
+//go:generate mapper method -i -d github.com/canonical/microcluster/cluster -e DiskPath GetMany
+//go:generate mapper method -i -d github.com/canonical/microcluster/cluster -e DiskPath GetOne
+//go:generate mapper method -i -d github.com/canonical/microcluster/cluster -e DiskPath ID
+//go:generate mapper method -i -d github.com/canonical/microcluster/cluster -e DiskPath Exists
+//go:generate mapper method -i -d github.com/canonical/microcluster/cluster -e DiskPath Create
+//go:generate mapper method -i -d github.com/canonical/microcluster/cluster -e DiskPath DeleteOne-by-Member-and-Path
+//go:generate mapper method -i -d github.com/canonical/microcluster/cluster -e DiskPath DeleteMany-by-Member
+//go:generate mapper method -i -d github.com/canonical/microcluster/cluster -e DiskPath Update
 
-// Disk is used to track the Ceph disks on a particular server.
-type Disk struct {
+// DiskPath is used to track the Ceph disks on a particular server.
+type DiskPath struct {
```

Comment:
One issue with using a very specific name such as DiskPath for the Disks table struct is that in future if we extend OSD functionality which requires additional columns in the table, the name `DiskPath` might not be relevant for the struct then. Which would mean another sql migration going forward ? Since the db change is basically dropping a (column in table/ field in struct) I suggest we keep the `Disk` name and not change it ?